instruments/abstract_instruments/comm/visa_communicator.py

```diff
@@ -29,9 +29,6 @@ class VisaCommunicator(io.IOBase, AbstractCommunicator):
     def __init__(self, conn):
         super(VisaCommunicator, self).__init__(self)
 
-        if pyvisa is None:
-            raise ImportError("PyVISA required for accessing VISA instruments.")
-
```

Comment:
Not sure this can ever be triggered, wouldn't it throw an error during the `import pyvisa` above if `pyvisa` is not installed?

Reply:
Indeed it would. Also, pyvisa is named as a dependency of IK, so I'm not sure in what scenario your users would end up with pyvisa not available?

In any case, I think the normal pattern for "optional" packages is doing something like this above:
```
try:
    import pyvisa
except ImportError:
    pyvisa = None
```
, then this would work as you expect (I think).

Reply:
In addition, just noticed that the same check is also in `instrument.py` before opening the visa communicator alltogether. the import there is the same, i.e., `import pyvisa`. Probably makes not much sense there either I'd assume...

Reply:
Its an old code block from way back when it was an optional dep, because installation required a separate non-Python package to be installed.